Source/Dafny/Verifier/Translator.DataTypes.cs

```diff
@@ -16,9 +16,7 @@ partial class Translator {
       Contract.Requires(dt != null);
       Contract.Requires(sink != null && predef != null);
 
-      foreach (DatatypeCtor ctor in dt.Ctors) {
-        AddDataTypeConstructor(dt, ctor);
-      }
+      var ctorFunctions = dt.Ctors.ToDictionary(c => c, c => AddDataTypeConstructor(dt, c));
```

Comment:
The variable `ctorFunction is used nowhere. Is there a good reason to build a collection here (a Dictionary all the more), instead of the original foreach loop?